
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Standalone version to add custom genome sets &#8212; GCB_MAN  documentation</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Use cases" href="usecases.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="usecases.html" title="Use cases"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">GCB_MAN  documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="standalone-version-to-add-custom-genome-sets">
<h1>Standalone version to add custom genome sets<a class="headerlink" href="#standalone-version-to-add-custom-genome-sets" title="Permalink to this headline">¶</a></h1>
<p>GCB can be used in three main ways: 1) as a web server with a precalculated set of genomes; 2) standalone application with browser-based GUI; 3) set of command-line scripts.</p>
<p>Standalone version should be used when the user wants to work with a custom set of genomes. Command-line scripts are provided to: calculate complexity profile, generate subgraphs, generate a database which can be imported to browser-based GUI application. Scheme of actions and scripts is shown below.</p>
<img alt="_images/scheme.png" class="align-center" src="_images/scheme.png" />
<div class="section" id="orthology-group-inference">
<h2>Orthology group inference<a class="headerlink" href="#orthology-group-inference" title="Permalink to this headline">¶</a></h2>
<p>To add a custom set of genomes orthology groups should be inferred first. We provide snakemake script: <a class="reference external" href="https://github.com/paraslonic/orthosnake">https://github.com/paraslonic/orthosnake</a> to accomplish this task. It takes fasta formatted genome sequences as input. Then gene annotation with prokka  tool of each genome is performed. Genbank files then converted to fasta formatted amino acid protein sequences with a custom python3 script. This script inserts special information about genes in fasta headers, namely: genome file name, numeric id, product name, contig, start, end (for example, <code class="docutils literal notranslate"><span class="pre">&gt;GCF_000007445|4|Threonine_synthase|NC_004431.1|4445|5731</span></code>). Then these files are used to infer orthology groups with OrthoFinder tool. The resulting file with orthology groups (OG) contains information about each OG in the following format: <code class="docutils literal notranslate"><span class="pre">&lt;og</span> <span class="pre">id&gt;:</span> <span class="pre">&lt;gene1&gt;</span> <span class="pre">&lt;gene2&gt;</span> <span class="pre">…</span></code></p>
<p>For example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">OG0008594</span><span class="p">:</span> <span class="n">GCF_001618325</span><span class="o">|</span><span class="mi">2406</span><span class="o">|</span><span class="n">Small_toxic_polypeptide_LdrD</span><span class="o">|</span><span class="n">NZ_CP015069</span><span class="o">.</span><span class="mi">1</span><span class="o">|</span><span class="mi">2607133</span><span class="o">|</span><span class="mi">2607240</span> <span class="n">GCF_001663475</span><span class="o">|</span><span class="mi">366</span><span class="o">|</span><span class="n">Small_toxic_polypeptide_LdrD</span><span class="o">|</span><span class="n">NZ_CP015159</span><span class="o">.</span><span class="mi">1</span><span class="o">|</span><span class="mi">380042</span><span class="o">|</span><span class="mi">380149</span>
</pre></div>
</div>
</div>
<div class="section" id="generating-of-the-graph-structure">
<h2>Generating of the graph structure<a class="headerlink" href="#generating-of-the-graph-structure" title="Permalink to this headline">¶</a></h2>
<p>When orthology groups are inferred, the next step is parsing of Orthofinder outputs. To do this you should open source directory and type in terminal:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python3</span> <span class="n">orthofinder_parse</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">i</span> <span class="p">[</span><span class="n">path</span> <span class="n">to</span> <span class="n">txt</span> <span class="n">file</span> <span class="k">with</span> <span class="n">orthogroups</span><span class="p">]</span> <span class="o">-</span><span class="n">o</span> <span class="p">[</span><span class="n">path</span> <span class="ow">and</span> <span class="n">name</span> <span class="n">prefix</span> <span class="k">for</span> <span class="n">output</span> <span class="n">files</span><span class="p">]</span>
</pre></div>
</div>
<p>For example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python3</span> <span class="n">orthofinder_parse</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">i</span> <span class="o">~/</span><span class="n">data</span><span class="o">/</span><span class="n">Mycoplasma</span><span class="o">/</span><span class="n">Results</span><span class="o">/</span><span class="n">Orthogroups</span><span class="o">.</span><span class="n">txt</span> <span class="o">-</span><span class="n">o</span> <span class="o">~/</span><span class="n">data</span><span class="o">/</span><span class="n">outputs</span><span class="o">/</span><span class="n">Mycoplasma</span><span class="o">/</span><span class="n">graph</span>
</pre></div>
</div>
<p>Output files:</p>
<blockquote>
<div><ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">graph.sif</span></code> - all edges list of the genomes graph</p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">graph.db</span></code> - SQLite database with all parsed information</p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">graph_context.sif</span></code> - number of unique contexts, computed for each node in the graph</p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">graph_genes.sif</span></code> - list of all genes (nodes) from all genomes, with coordinates and Prokka annotations</p></li>
</ul>
</div></blockquote>
<p>The main graph structure is stored in a text graph.sif file, where each one line describes one edge, with its source and target nodes, genome id and contig id, to which this edge belongs. This file is used to create a graph object, which is used in all graph processing procedures.</p>
</div>
<div class="section" id="complexity-computing">
<h2>Complexity computing<a class="headerlink" href="#complexity-computing" title="Permalink to this headline">¶</a></h2>
<p>The next step is the computing of genome complexity. To do this type in terminal:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python3</span> <span class="n">start_computing</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">i</span> <span class="n">graph</span><span class="o">.</span><span class="n">sif</span> <span class="o">-</span><span class="n">o</span> <span class="p">[</span><span class="n">path</span> <span class="n">to</span> <span class="n">output</span> <span class="n">folder</span><span class="p">]</span> <span class="o">--</span><span class="n">reference</span> <span class="p">[</span><span class="n">name</span> <span class="n">of</span> <span class="n">reference</span> <span class="n">genome</span><span class="p">]</span>
</pre></div>
</div>
<dl>
<dt>Additional parameters:</dt><dd><p>–window - sliding window size (default 20)</p>
<p>–iterations - number of iterations in probabilistic method (default 500)</p>
<p>–genomes_list - path to file with a list of names which will be used to create a graph (default all strains from <a href="#id1"><span class="problematic" id="id2">*</span></a>.sif will be used)</p>
<p>–min_depth, –max_depth - minimum and maximum depth of generated paths in the graph (default from 0 to inf)</p>
<p>–save_db - path to the database, created by orthfinder_parse.py (default data will not be saved to db, only to txt). It’s necessary to use this parameter if you want to use this complexity profile in the stand-alone browser-based GCB application.</p>
</dd>
</dl>
<p>Output files for each contig in the reference genome:</p>
<blockquote>
<div><p><code class="file docutils literal notranslate"><span class="pre">all_bridges_contig_n.txt</span></code> - this file contains information about the number of deviating paths between each pair of nodes in the reference genome</p>
</div></blockquote>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Standalone version to add custom genome sets</a><ul>
<li><a class="reference internal" href="#orthology-group-inference">Orthology group inference</a></li>
<li><a class="reference internal" href="#generating-of-the-graph-structure">Generating of the graph structure</a></li>
<li><a class="reference internal" href="#complexity-computing">Complexity computing</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="usecases.html"
                        title="previous chapter">Use cases</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/standalone.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="usecases.html" title="Use cases"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">GCB_MAN  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2020, ADD.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.2.0.
    </div>
  </body>
</html>