
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Use cases &#8212; GCB_MAN  documentation</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Standalone version to add custom genome sets" href="standalone.html" />
    <link rel="prev" title="Interface elements and settings" href="interface.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="standalone.html" title="Standalone version to add custom genome sets"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="interface.html" title="Interface elements and settings"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">GCB_MAN  documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="use-cases">
<h1>Use cases<a class="headerlink" href="#use-cases" title="Permalink to this headline">¶</a></h1>
<div class="section" id="determine-context-and-variants-of-operon-of-interest">
<h2>Determine context and variants of operon of interest<a class="headerlink" href="#determine-context-and-variants-of-operon-of-interest" title="Permalink to this headline">¶</a></h2>
<p>Consider we are interested in a particular operon and want to know more about its representation in a particular group of organisms. Let’s take <a class="reference external" href="https://en.wikipedia.org/wiki/Lac_operon">lactose operon</a> with its regulator in the Escherichia coli as an example (we will call it lac operon further).</p>
<p>First we need to determine genome position of genes of interest. We can do it in multiple ways, i.e. from <a class="reference external" href="https://www.google.com/url?q=https://biocyc.org/ECOLI/NEW-IMAGE?type%3DOPERON%26object%3DTU00036&amp;sa=D&amp;ust=1585816672295000">EcoCyc</a> database or from <a class="reference external" href="https://www.ncbi.nlm.nih.gov/nuccore/NC_000913.3">NCBI Refseq</a>. In our case, according to BioCyc, genome region of interst is located at 361249-366305 in <em>K-12 substr. MG1655</em>. First we select Escherichia coli as <code class="docutils literal notranslate"><span class="pre">Organism</span></code> in a dataset containing 300 <em>E. coli</em> genomes. Then we select K-12 MG1655 as <code class="docutils literal notranslate"><span class="pre">Reference</span></code>. Only one contig is present (finished genome, no plasmids), so we do not need to change <code class="docutils literal notranslate"><span class="pre">Contig</span></code> value.</p>
<a class="reference internal image-reference" href="_images/use1_1.png"><img alt="_images/use1_1.png" src="_images/use1_1.png" style="width: 485.6px; height: 155.20000000000002px;" /></a>
<p>Then we should put the operon’s margin coordinates to text fields: <code class="docutils literal notranslate"><span class="pre">Start</span> <span class="pre">coordinate</span></code> and <code class="docutils literal notranslate"><span class="pre">End</span> <span class="pre">coordinate</span></code>.</p>
<a class="reference internal image-reference" href="_images/use1_2.png"><img alt="_images/use1_2.png" src="_images/use1_2.png" style="width: 685.6px; height: 70.4px;" /></a>
<p>Now we click <code class="docutils literal notranslate"><span class="pre">Draw</span></code> button on the bottom panel and graph representation of the operon appears.</p>
<img alt="_images/use1_3.png" src="_images/use1_3.png" />
<p>We can see that edges are of different colors: black and red. Red colored are edges which are in agreement with the reference genome, i.e. they represent the gene order of reference genome. All other edges are black, they correspond to genes neighbourhood variants not present in a reference genome. Nodes of genes which present in a reference genome are colored pink, nodes for genes absent in a reference genome are colored green. Gray rectangle is drawn around nodes located between <code class="docutils literal notranslate"><span class="pre">Start</span> <span class="pre">coordinate</span></code> and <code class="docutils literal notranslate"><span class="pre">End</span> <span class="pre">coordinate</span></code> (lac operon genes in our case), other nodes are added to show the context.</p>
<p>More clear layout can be obtained manually by left clicking and dragging nodes with mouse.</p>
<img alt="_images/use4_.png" src="_images/use4_.png" />
<p>We can see now that operon is located in a conservative context in most of the genomes (thick edges designated with 1 and 2).
We may notice that one of the operon’s genes is absent in some set of genomes. By left clicking on that gene we can examine its gene product. Gene products are assigned with <a class="reference external" href="https://github.com/tseemann/prokka">prokka</a> tool in the web version of GCB. Text in popup box is formated as <code class="docutils literal notranslate"><span class="pre">`product:</span> <span class="pre">length</span> <span class="pre">(position)`</span></code> (length in nucleotides, position is reported only for genes present in refernce genome).</p>
<img alt="_images/select_node.gif" src="_images/select_node.gif" />
<p>In our case it is Galactoside acetyltransferase which is missing from the operon in some set of genomes. What are that genomes? By clicking on the bypassing edge we can select it. Selection of the edge results in two effects: 1) names of genomes corresponding to this edge (those genomes in which linked by the edge genes are located next to each other) appears in a <code class="docutils literal notranslate"><span class="pre">List</span> <span class="pre">of</span> <span class="pre">genomes</span></code> section below the graph, 2) other edges containing at least one of the genomes corresponding to the selected edge are colored blue. In this way, we can determine what possible variants of gene content and gene order are present, and in which genomes.</p>
<img alt="_images/select_edge.gif" src="_images/select_edge.gif" />
<p>For now we have determined that a number of genomes does not contain Galactoside acetyltransferase (we can copy-paste their names and calculate the total number, 29 in this case). We can also notice nodes connected by a thin edges which seems to represent other alternative variants of the operon. Let’s click on them to see their products.</p>
<img alt="_images/look_orphans1.gif" src="_images/look_orphans1.gif" />
<p>To verify our findings let’s switch to paralogues orthologization mode. To do it you should toggle <code class="docutils literal notranslate"><span class="pre">Draw</span> <span class="pre">paralogous</span></code> switcher on the top panel and click <code class="docutils literal notranslate"><span class="pre">Draw</span></code> button once more (be careful, your current graph layout will be lost, so consider opening new page with GCB). After clicking and dragging nodes it should be something like this.</p>
<img alt="_images/paralogs.png" src="_images/paralogs.png" />
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Use cases</a><ul>
<li><a class="reference internal" href="#determine-context-and-variants-of-operon-of-interest">Determine context and variants of operon of interest</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="interface.html"
                        title="previous chapter">Interface elements and settings</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="standalone.html"
                        title="next chapter">Standalone version to add custom genome sets</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/usecases.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="standalone.html" title="Standalone version to add custom genome sets"
             >next</a> |</li>
        <li class="right" >
          <a href="interface.html" title="Interface elements and settings"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">GCB_MAN  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2020, ADD.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.2.0.
    </div>
  </body>
</html>