
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Frequently Asked Questions &#8212; Genome Complexity Browser  documentation</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Contacts" href="contacts.html" />
    <link rel="prev" title="Standalone version" href="standalone.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="contacts.html" title="Contacts"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="standalone.html" title="Standalone version"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">Genome Complexity Browser  documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="frequently-asked-questions">
<h1>Frequently Asked Questions<a class="headerlink" href="#frequently-asked-questions" title="Permalink to this headline">Â¶</a></h1>
<p>What are those circles and lines in the graph form?</p>
<p>How gene name is selected?</p>
<p>Does all variants that present in considered set of genomes are represented in the graph visualization form?</p>
<ul class="simple">
<li><p>Not always. GCB by default skips paralogues genes, but they can still be analysed by switching <code class="docutils literal notranslate"><span class="pre">Draw</span> <span class="pre">paralogues</span></code> toggle on top panel.</p></li>
</ul>
<p>Where should I take coordinates of genes of interest?
- from genome annotations (i.e. Genebank or RefSeq database) by searching for particular gene.
- DOOR database [Mao, Xizeng, et al., Nucleic acids research, 2014] (currently available <a class="reference external" href="http://161.117.81.224/DOOR3/">here</a> ) can be used to find information about operons of specific organism.
- from papers, in which genome coordinates are specified.</p>
<p>How can I obtain graph image in vector format for publication?
- Export it to JSON and then open it in a Cytoscape or other graph editing software.</p>
<p>What if genome of interest is absent at gcb.rcpcm.org?</p>
<p>Can I use PC for the standalone analyisis or do I need a computational cluster?</p>
<p>How many genomes may be analyzed in a standalone version?</p>
<p>Can I adress someone to help me with my organisms?</p>
<ul class="simple">
<li><p>You can write to the <a class="reference external" href="https://groups.google.com/forum/#!forum/genome-complexiity-browser">google group</a>. We will be hapy to help you.</p></li>
</ul>
<p>Why should I run several python scripts insted of one in a standalone version?
- you may choose different references. Only complexity values for the selected reference are added to database. Usually we have one or several genomes of particular interest or of good quality which may become references. Computation of complexity values for each genome in a set to</p>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h4>Previous topic</h4>
  <p class="topless"><a href="standalone.html"
                        title="previous chapter">Standalone version</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="contacts.html"
                        title="next chapter">Contacts</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/FAQ.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="contacts.html" title="Contacts"
             >next</a> |</li>
        <li class="right" >
          <a href="standalone.html" title="Standalone version"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">Genome Complexity Browser  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2020, A. Manolov, D. Konanov, D. Fedorov, R. Vereshchagin, E. Ilina.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.2.0.
    </div>
  </body>
</html>