
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>GUI description &#8212; Genome Complexity Browser  documentation</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Standalone version" href="standalone.html" />
    <link rel="prev" title="Main concepts" href="general.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="standalone.html" title="Standalone version"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="general.html" title="Main concepts"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">Genome Complexity Browser  documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="gui-description">
<h1>GUI description<a class="headerlink" href="#gui-description" title="Permalink to this headline">¶</a></h1>
<div class="section" id="gui-elements">
<h2>GUI elements<a class="headerlink" href="#gui-elements" title="Permalink to this headline">¶</a></h2>
<p>GCB page consists of three main parts:</p>
<blockquote>
<div><ul>
<li><p>Complexity plot which shows complexity profile for selected genome and contig</p></li>
<li><p>Subgraph visualization of selected genome region</p></li>
<li><p>Left sliding panel to:</p>
<blockquote>
<div><ul class="simple">
<li><p>select genome and region to work with</p></li>
<li><p>search gene product names</p></li>
<li><p>download complexity values and graph.</p></li>
</ul>
</div></blockquote>
</li>
</ul>
</div></blockquote>
<a class="reference internal image-reference" href="_images/annotated_screen.png"><img alt="_images/annotated_screen.png" class="align-center" src="_images/annotated_screen.png" style="width: 504.8px; height: 560.8000000000001px;" /></a>
</div>
<div class="section" id="tutorials">
<h2>Tutorials<a class="headerlink" href="#tutorials" title="Permalink to this headline">¶</a></h2>
<div class="section" id="complexity-analysis">
<h3>Complexity analysis<a class="headerlink" href="#complexity-analysis" title="Permalink to this headline">¶</a></h3>
<p>Complexity is the measure of local genome variability. It is calculated for a set of genomes, one of which is selected as a reference. The greater the number of local changes in a certain neighborhood, the greater the value of complexity. Locality is set by the parameter <code class="docutils literal notranslate"><span class="pre">Window</span></code>, the larger it is, the more large-scale changes influences complexity value.</p>
<p>Thus, the three parameters that you need to select to calculate the complexity profile are: 1) set of genomes, 2) reference genome, 3) window size. All of them can be set in the left sliding panel.</p>
<p>At gcb.rcpcm.org we precalculated complexity profiles for over 140 prokaryotic species. Any set of genomes may be used in a stand-alone version. Window size equals 20 by default, and can be changed to other precalculated values (20, 50 and 100 values are available at gcb.rcpcm.org, any value may be used in a standalone version).</p>
<p>To obtain the complexity profile for a certain set of genomes (we will consider the set of <em>Bacillus Subtilus</em> available at gcb.rcpcm.org), follow these steps:</p>
<ol class="arabic simple">
<li><p>Open gcb.rcpcm.org in you web-browser.</p></li>
</ol>
<blockquote>
<div><p>Left panel with a number of settings will open automatically.</p>
</div></blockquote>
<ol class="arabic simple" start="2">
<li><p>Click on the <code class="docutils literal notranslate"><span class="pre">Organism</span></code> selector and choose <code class="docutils literal notranslate"><span class="pre">Bacillus_subtilis</span></code> item.</p></li>
<li><p>Click on the <code class="docutils literal notranslate"><span class="pre">Reference</span></code> selector and choose one of the available genomes. Let’s choose for example <code class="docutils literal notranslate"><span class="pre">GCF_001889385.1_ASM188938v1_genomic</span> <span class="pre">(J-5)</span></code> genome.</p></li>
</ol>
<blockquote>
<div><p>Genome name is constructed based on its filename (NCBI RefSeq filenames are used in the web version, genomes were downloaded from NCBI FTP server). In the web version strain name from the RefSeq is added to this name in brackets.</p>
</div></blockquote>
<ol class="arabic simple" start="4">
<li><p>Click on the <code class="docutils literal notranslate"><span class="pre">Reference</span></code> selector.</p></li>
</ol>
<blockquote>
<div><p>The contigs of the selected genome will be shown. For complete (finished) genomes they corresponds to replicones (chromosome(s) and plasmids), for draft genomes they are some fragments of genome. You cannot see the complexity profile for all contigs at the same time, they must be selected in turn.</p>
<p>In case of <code class="docutils literal notranslate"><span class="pre">J-5</span></code> genome, only one contig is present (<code class="docutils literal notranslate"><span class="pre">NZ_CP018295.1</span></code>), which means that it is a complete (finished) genome, with single chromosome and without plasmids.</p>
</div></blockquote>
<ol class="arabic simple" start="5">
<li><p>Close left sliding panel by clicking on the some point outside it (e.g., on the middle of the page).</p></li>
</ol>
<blockquote>
<div><p>To open this panel once again, click on the icon with three horizontal lines on the left side of the page.</p>
</div></blockquote>
<ol class="arabic simple" start="6">
<li><p>Complexity profile wile appear in the <em>Complexity Plot</em> panel.</p></li>
</ol>
<p>A number of peaks and flat areas are visible in the complexity profile. Peaks corresponds to the genome regions in which many changes were fixed during evolution.</p>
<ol class="arabic simple" start="7">
<li><p>To get the numerical values of the complexity profile, open left sliding panel, select the <em>File</em> tab at the top of the panel, click <code class="docutils literal notranslate"><span class="pre">Download</span> <span class="pre">complexity</span> <span class="pre">values</span></code> button.</p></li>
</ol>
<p><strong>Screen recording</strong> with these steps performed is available <a class="reference external" href="https://youtu.be/q122j3pbcko">here</a>.</p>
</div>
<div class="section" id="gene-neigborhood-analysis">
<h3>Gene neigborhood analysis<a class="headerlink" href="#gene-neigborhood-analysis" title="Permalink to this headline">¶</a></h3>
<p>Consider we are interested in a particular operon and we want to know more about its representation in a particular group of organisms. Let’s take <a class="reference external" href="https://en.wikipedia.org/wiki/Lac_operon">lactose operon</a> with its regulator in the Escherichia coli as an example (we will call it lac operon further).</p>
<p>First, we need to determine genome position of genes of interest. We can do it in multiple ways, e.g., from <a class="reference external" href="https://www.google.com/url?q=https://biocyc.org/ECOLI/NEW-IMAGE?type%3DOPERON%26object%3DTU00036&amp;sa=D&amp;ust=1585816672295000">EcoCyc</a> database or from <a class="reference external" href="https://www.ncbi.nlm.nih.gov/nuccore/NC_000913.3">NCBI Refseq</a>. According to EcoCyc, the operon is located at 361249-366305 in <em>K-12 substr. MG1655</em>. Let’s first select <em>Escherichia coli</em> as <code class="docutils literal notranslate"><span class="pre">Organism</span></code>, in a dataset containing 300 <em>E. coli</em> genomes. Then, select K-12 MG1655 as <code class="docutils literal notranslate"><span class="pre">Reference</span></code>. Only one contig is present (finished genome, no plasmids), so we do not need to change <code class="docutils literal notranslate"><span class="pre">Contig</span></code> value.</p>
<a class="reference internal image-reference" href="_images/use1_1.png"><img alt="_images/use1_1.png" src="_images/use1_1.png" style="width: 485.6px; height: 155.20000000000002px;" /></a>
<p>Then we should put the operon’s margin coordinates to text fields: <code class="docutils literal notranslate"><span class="pre">Start</span> <span class="pre">coordinate</span></code> and <code class="docutils literal notranslate"><span class="pre">End</span> <span class="pre">coordinate</span></code>.</p>
<a class="reference internal image-reference" href="_images/use1_2.png"><img alt="_images/use1_2.png" src="_images/use1_2.png" style="width: 685.6px; height: 70.4px;" /></a>
<p>Now click <code class="docutils literal notranslate"><span class="pre">DRAW</span> <span class="pre">GRAPH</span></code> button and graph-based representation of the operon will appear in <em>Graph</em> panel.</p>
<img alt="_images/use1_3.png" src="_images/use1_3.png" />
<p>Gray rectangle is drawn around nodes located between <code class="docutils literal notranslate"><span class="pre">Start</span> <span class="pre">coordinate</span></code> and <code class="docutils literal notranslate"><span class="pre">End</span> <span class="pre">coordinate</span></code> (lac operon genes in this case).</p>
<p>Note that edges are of different colors. Red edges are from the reference genome (and also from genomes in which genes are located in the same order), black edges correspond to variants not present in the reference genome.</p>
<p>Nodes also have different colors. Nodes of genes from the reference genome are colored red, if they are located inbetween <code class="docutils literal notranslate"><span class="pre">Start</span> <span class="pre">coordinate</span> <span class="pre">-</span> <span class="pre">neigbhourhood</span></code> and <code class="docutils literal notranslate"><span class="pre">End</span> <span class="pre">coordinate</span> <span class="pre">+</span> <span class="pre">neigbhourhood</span></code>; they are colored pink, if present in the reference genome, but located outside the mentioned region. Nodes of genes, that are absent in a reference genome, are colored green.</p>
<p>More clear layout can be obtained manually by left clicking and dragging nodes with mouse.</p>
<img alt="_images/use4_.png" src="_images/use4_.png" />
<p>By looking at thick edges designated with 1 and 2, we can tell that operon is located in a conserved context, in most of the genomes.</p>
<p>One of the operon’s genes is absent in some set of genomes. By left clicking on that gene, we can select it, and know its gene product (gene products are assigned with <a class="reference external" href="https://github.com/tseemann/prokka">Prokka</a>).</p>
<img alt="_images/select_node.gif" src="_images/select_node.gif" />
<p>Missing gene is Galactoside acetyltransferase. This gene is not in operons of a number of genomes. what are that genomes? Let’s click on bypassing edge.</p>
<img alt="_images/select_edge.gif" src="_images/select_edge.gif" />
<p>Click on the edge, selects it.</p>
<p>Selection of the edge results in two effects:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>names of genomes corresponding to this edge appears in the <code class="docutils literal notranslate"><span class="pre">List</span> <span class="pre">of</span> <span class="pre">genomes</span></code> section below the graph</p></li>
<li><p>other edges, that contain at least one of the genomes from the selected edge, are colored blue. In this way, one can determine possible variants of gene neigbhorhoods, and in which genomes they are present.</p></li>
</ol>
</div></blockquote>
<p>For now we have determined, that a number of genomes does not contain Galactoside acetyltransferase. We can also notice nodes connected by a thin edges, which seems to represent other alternative variants of the operon. Let’s click on that nodes and on the nodes from the reference, to see their products.</p>
<img alt="_images/look_orphans1.gif" src="_images/look_orphans1.gif" />
<p>We see that their names are the same, but their length differs a lot: 263 b.p. for an outlier gene and 1253 b.p. for a reference gene. Often, and also in this particular case,  it comes from frameshit splitting some genes into parts, some of which may become part of homology groups representing original gene.</p>
<img alt="_images/look_orphans1.gif" src="_images/look_orphans1.gif" />
<p>Finaly to verify our findings let’s switch to paralogues orthologization mode. To do it you should toggle <code class="docutils literal notranslate"><span class="pre">Draw</span> <span class="pre">paralogous</span></code> switcher on the top panel and click <code class="docutils literal notranslate"><span class="pre">Draw</span></code> button once more (be careful, your current graph layout will be lost, so consider opening new page). After clicking and dragging nodes it should be looking like this. A little bit scary.</p>
<img alt="_images/paralogs.png" src="_images/paralogs.png" />
<p>This more complicated graph comes from not ignoring paralogous genes as it done by default, but instead showing all of them.</p>
</div>
<div class="section" id="combined-analysis">
<h3>Combined analysis<a class="headerlink" href="#combined-analysis" title="Permalink to this headline">¶</a></h3>
<p>With GCB, you can find which genes are in the hot spots of genome variability.</p>
<p>To do this, first select an organism, strain and replicon (chromosome or plasmids, complete genomes are reccomended to be used as reference).</p>
<p>Then, in complexity profile panel, click on some of the hot spots to set the current position. Current position is marked with vertical line in the complexity plot and also in <code class="docutils literal notranslate"><span class="pre">Start</span></code> and <code class="docutils literal notranslate"><span class="pre">End</span></code> coordinates in the left sliding panel).</p>
<p>Before proceding to the graph visualization, we recommend adjusting graph rendering options: set <code class="docutils literal notranslate"><span class="pre">Minimal</span> <span class="pre">edge</span> <span class="pre">value</span></code> to 10 (the more intense the hostpot, the bigger this value should be), <code class="docutils literal notranslate"><span class="pre">Window</span></code> to 10-20, depending on the hotspot width.  Now press the <code class="docutils literal notranslate"><span class="pre">DRAW</span> <span class="pre">GRAH</span></code> button in the upper left corner of the <em>Complexity</em> panel. Changing colors will be visible above the graph draw buttons while it is being built, and then graph will apear in the <em>Grpah</em> panel.</p>
<p>To select some genes, for example, located at the variable region, press the left mouse button and while holding it move the cursor to surround the desired genes. Their products will apear in the bottom right part of the <strong>Graph</strong> panel.</p>
<p><strong>Screen recording</strong> with these steps, and medium hot spot, is available <a class="reference external" href="https://youtu.be/q122j3pbcko">here</a>.</p>
</div>
<div class="section" id="publication-ready-graph-rendering">
<h3>Publication-ready graph rendering<a class="headerlink" href="#publication-ready-graph-rendering" title="Permalink to this headline">¶</a></h3>
<p>A graph-based representation of genome region can be exported in the form of JPEG image or a JSON file. To do this, fist draw some graph and then go to the left sliding panel, select <em>File</em> tab, select <em>Graph</em> subtab, click <code class="docutils literal notranslate"><span class="pre">DOWNLOAD</span> <span class="pre">JPEG</span></code> or <code class="docutils literal notranslate"><span class="pre">DOWNLOAD</span> <span class="pre">JSON</span></code> buttons.</p>
<p>JPEG file stores only a bitmap image, JSON file contains all infromation regarding the current graph, including its layout.</p>
<p>JSON file can be imported in <a class="reference external" href="https://cytoscape.org/">Cytoscape</a> for complete visualization control (customize the look of nodes, edges, do manual or one of the automatic layouts). Cytoscape graph renderings can be exported in a number of bitmap and vector formats (e.g., pdf, svg).</p>
<p>To import JSON graph file into Cytoscape, select <code class="docutils literal notranslate"><span class="pre">File</span></code> - <code class="docutils literal notranslate"><span class="pre">Import</span></code> - <code class="docutils literal notranslate"><span class="pre">Network</span> <span class="pre">from</span> <span class="pre">File</span></code> and select file, that was downloaded from GCB. Now you may arange graph nodes and adjust style.</p>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">GUI description</a><ul>
<li><a class="reference internal" href="#gui-elements">GUI elements</a></li>
<li><a class="reference internal" href="#tutorials">Tutorials</a><ul>
<li><a class="reference internal" href="#complexity-analysis">Complexity analysis</a></li>
<li><a class="reference internal" href="#gene-neigborhood-analysis">Gene neigborhood analysis</a></li>
<li><a class="reference internal" href="#combined-analysis">Combined analysis</a></li>
<li><a class="reference internal" href="#publication-ready-graph-rendering">Publication-ready graph rendering</a></li>
</ul>
</li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="general.html"
                        title="previous chapter">Main concepts</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="standalone.html"
                        title="next chapter">Standalone version</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/interface.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="standalone.html" title="Standalone version"
             >next</a> |</li>
        <li class="right" >
          <a href="general.html" title="Main concepts"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">Genome Complexity Browser  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2020, A. Manolov, D. Konanov, D. Fedorov, R. Vereshchagin, E. Ilina.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.2.0.
    </div>
  </body>
</html>