
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Standalone version &#8212; Genome Complexity Browser  documentation</title>
    <link rel="stylesheet" href="_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/language_data.js"></script>
    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Frequently Asked Questions" href="FAQ.html" />
    <link rel="prev" title="GUI description" href="interface.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="FAQ.html" title="Frequently Asked Questions"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="interface.html" title="GUI description"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">Genome Complexity Browser  documentation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="standalone-version">
<h1>Standalone version<a class="headerlink" href="#standalone-version" title="Permalink to this headline">¶</a></h1>
<div class="section" id="command-line-tool">
<h2>Command-line tool<a class="headerlink" href="#command-line-tool" title="Permalink to this headline">¶</a></h2>
<div class="section" id="roadmap">
<h3>Roadmap<a class="headerlink" href="#roadmap" title="Permalink to this headline">¶</a></h3>
<p>Standalone version should be used when the user wants to work with a custom set of genomes.
Command-line scripts are provided to: calculate complexity profile, generate subgraphs, generate a database which can be imported to browser-based GUI application. Scheme of actions and scripts is shown below.</p>
<img alt="_images/standalone_scheme_hor_short.png" class="align-center" src="_images/standalone_scheme_hor_short.png" />
</div>
<div class="section" id="prerequisites">
<h3>Prerequisites<a class="headerlink" href="#prerequisites" title="Permalink to this headline">¶</a></h3>
<p><strong>Conda</strong></p>
<p>Please refer to the <a class="reference external" href="https://docs.conda.io/projects/conda/en/latest/user-guide/install/">installation guide</a>. Select Python3 version.</p>
<p><strong>Snakemake</strong></p>
<p>Installation instrctions are <a class="reference external" href="https://snakemake.readthedocs.io/en/stable/getting_started/installation.html">here</a>.</p>
<p><strong>Python 3.4 or later</strong></p>
<p>Comes with conda.</p>
<p><strong>Graphviz and pygraphviz libraries</strong></p>
<p>Can be installed by running:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sudo</span> <span class="n">apt</span><span class="o">-</span><span class="n">get</span> <span class="n">install</span> <span class="n">graphviz</span> <span class="n">python3</span><span class="o">-</span><span class="n">graphviz</span> <span class="n">python3</span><span class="o">-</span><span class="n">pygraphviz</span>
</pre></div>
</div>
<p>or via conda:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">c</span> <span class="n">anaconda</span> <span class="n">graphviz</span>
<span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">c</span> <span class="n">anaconda</span> <span class="n">pygraphviz</span>
</pre></div>
</div>
<p><strong>gene_graph_lib Python3 module</strong></p>
<p>Can be installed by running:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">pip3</span> <span class="n">install</span> <span class="n">gene_graph_lib</span>
</pre></div>
</div>
</div>
<div class="section" id="orthogroup-inference">
<h3>Orthogroup inference<a class="headerlink" href="#orthogroup-inference" title="Permalink to this headline">¶</a></h3>
<p>Orthogroup inference is the first step in the standalone analysis.
We recomend using our <a class="reference external" href="https://github.com/paraslonic/orthosnake">orthosnake</a> pipeline for inference of orthogroups, because GCB requires some special formating of the files.</p>
<p><strong>Orthosnake pipeline</strong></p>
<p><em>INPUT: Fasta-formated files with .fna extension, one file per genome</em></p>
<p><em>OUTPUT: orthogroups file `Orthogroups.txt` in OrthoFinder format</em></p>
<p>Steps:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>Clone or download orthosnake GIT repository: <a class="reference external" href="https://github.com/paraslonic/orthosnake">https://github.com/paraslonic/orthosnake</a></p></li>
<li><p>Put fasta-formated genome files in <code class="docutils literal notranslate"><span class="pre">fna</span></code> folder of the orthosnake folder.</p></li>
<li><p>Rut snakemake with <code class="docutils literal notranslate"><span class="pre">--use-conda</span></code> and other appropriate options</p></li>
</ol>
</div></blockquote>
<p>The following code will run analysis of the test dataset (three plasmids):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">paraslonic</span><span class="o">/</span><span class="n">orthosnake</span><span class="o">.</span><span class="n">git</span>
<span class="n">cd</span> <span class="n">orthosnake</span>
<span class="n">cp</span> <span class="n">test_fna</span><span class="o">/*</span> <span class="n">fna</span> <span class="c1"># copy test fasta files with three plasmids</span>
<span class="n">snakemake</span> <span class="o">-</span><span class="n">j</span> <span class="mi">4</span> <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">conda</span>
</pre></div>
</div>
<p>Here, we used snakemake with parameters specifying the number of available cores (<code class="docutils literal notranslate"><span class="pre">-j</span> <span class="pre">4</span></code>), and using conda environments (<code class="docutils literal notranslate"><span class="pre">--use-conda</span></code>). During the first start of the pipeline, it will take about ten minutes to install the necessary programs into the conda environments.</p>
<p>If genome files have extension, other than <code class="docutils literal notranslate"><span class="pre">.fna</span></code> (e.g., <code class="docutils literal notranslate"><span class="pre">*.fasta</span></code>) , please rename them. For example, to change the file extension from <code class="docutils literal notranslate"><span class="pre">.fasta</span></code> to <code class="docutils literal notranslate"><span class="pre">.fna</span></code>, run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>for i in fna/*.fasta; do mv $i fna/$(basename $i .fasta).fna; done
</pre></div>
</div>
<p>Orthosnake pipeline performs the following steps:</p>
<blockquote>
<div><ul class="simple">
<li><dl class="simple">
<dt>Fasta headers are modified, to satisfy the requirement of the Prokka:</dt><dd><ul>
<li><p>symbols other than alphanumericals and <cite>_</cite> are converted to <cite>_</cite></p></li>
<li><p>if header is longer than 20 symbols, it is cropped to the first 18 symbols, and dots are added to the end (e.g., <code class="docutils literal notranslate"><span class="pre">gi|15829254|ref|NC_002695.1</span></code> becomes <code class="docutils literal notranslate"><span class="pre">gi|15829254|ref|NC..</span></code>)</p></li>
</ul>
</dd>
</dl>
</li>
<li><p>Annotation with Prokka.</p></li>
<li><p>Genebank files converted to amino acid fasta files.</p></li>
<li><p>Orthogroups are inferred with OrthoFinder.</p></li>
</ul>
</div></blockquote>
<p>When amino acid fasta files are generated from genebank files, information regarding location and product is included in headers in the following format: genome_name|numerical_id|start|end|product. Please note this if you want to perform orthogroups inference without using orthosnake.</p>
</div>
<div class="section" id="building-a-graph-and-complexity-estimation-with-a-single-command">
<h3>Building a graph and complexity estimation with a single command<a class="headerlink" href="#building-a-graph-and-complexity-estimation-with-a-single-command" title="Permalink to this headline">¶</a></h3>
<p><em>INPUT: orthogroups file `Orthogroups.txt` in OrthoFinder format</em></p>
<p><em>OUTPUT: graph-based representation of genomes (sqlite database), complexity values (text file)</em></p>
<p>To perform the basic analysis, run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python3</span> <span class="n">gg</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">i</span> <span class="p">[</span><span class="n">orthogroups</span> <span class="n">file</span><span class="p">]</span> <span class="o">-</span><span class="n">o</span> <span class="p">[</span><span class="n">output</span><span class="p">]</span> <span class="o">--</span><span class="n">reference</span> <span class="p">[</span><span class="n">name</span> <span class="n">of</span> <span class="n">the</span> <span class="n">reference</span> <span class="n">genome</span><span class="p">]</span>
</pre></div>
</div>
<p>Parameters:</p>
<blockquote>
<div><p><code class="docutils literal notranslate"><span class="pre">-i</span></code> – path to orthogroups file in OrthoFinder format.</p>
<p><code class="docutils literal notranslate"><span class="pre">-o</span></code> –       path and name prefix for output files</p>
<p><code class="docutils literal notranslate"><span class="pre">--reference</span></code> –      name of the reference genome (its filename without extension), used for complexity estimation. This parameter can be skipped, in this case, the complexity values for all genomes in the set will be estimated (this can take a lot of time).</p>
</div></blockquote>
<p>Optional parameters:</p>
<blockquote>
<div><p><code class="docutils literal notranslate"><span class="pre">--window</span></code>  – size of the window used in the complexity estimation (default is 20). Only the paths inside the window increase the value of complexity.</p>
<p><code class="docutils literal notranslate"><span class="pre">--genomes_list</span></code> – a text file containing the names of the genomes that will be used to calculate the complexity. Useful for analysis of parts of genomes, such as phylogenetic tree clades, for their subsequent comparison.</p>
<p><code class="docutils literal notranslate"><span class="pre">--coalign</span></code> – a binary value (True/False) that determines whether to perform the step of selecting the optimal genomes orientation. It can take a lot of time when using a large number of draft genomes (5 hours for 1000 draft genomes).</p>
</div></blockquote>
<p>Advanced complexity estimation algorithm settings (practically not used):</p>
<blockquote>
<div><p><code class="docutils literal notranslate"><span class="pre">--iterations</span></code> –  number of iterations in algorithm (default is 500)</p>
<p><code class="docutils literal notranslate"><span class="pre">--min_depth</span></code> – minimum length of deviating path (default is 0)</p>
<p><code class="docutils literal notranslate"><span class="pre">--max_depth</span></code> – maximum length of deviating path (default is inf)</p>
</div></blockquote>
<p>What this command does:</p>
<blockquote>
<div><ul class="simple">
<li><p>parses <cite>OrthoGroups.txt</cite></p></li>
<li><p>creates graph file in sif-format and additional information about genes, their contextes, etc.</p></li>
<li><p>creates SQLite database used in GCB</p></li>
<li><p>computes complexity profiles for ALL genomes in the dataset and fills the DB</p></li>
<li><p>dumps graph object for fast access</p></li>
<li><p>all operations are executed by two ways: with deletion of paralogues, and with artificial orthologization</p></li>
</ul>
</div></blockquote>
<p>Main output files are:</p>
<blockquote>
<div><ul class="simple">
<li><p><code class="file docutils literal notranslate"><span class="pre">&lt;output&gt;.db</span></code> - SQLite database conatining graph and complexity values,  paralogues genes are skipped.</p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">&lt;output&gt;_pars.db</span></code> - SQLite database conatining graph and complexity values, paralogues genes are orthologized.</p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">[reference</span> <span class="pre">genome]/prob_window_complexity_contig_[contig].txt</span></code> - text file containing complexity values for each contig in the reference genome.</p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">&lt;output&gt;_context.sif</span></code> - number of unique contexts, computed for each node in graph</p></li>
<li><p><code class="file docutils literal notranslate"><span class="pre">&lt;output&gt;_genes.sif</span></code> - list of all genes (nodes) from all genomes, with coordinates and Prokka annotations</p></li>
</ul>
</div></blockquote>
<p>Where, <code class="docutils literal notranslate"><span class="pre">&lt;output&gt;</span></code> is what was specifiend in <code class="docutils literal notranslate"><span class="pre">-o</span></code> option, while running <cite>gg.py</cite>.</p>
</div>
</div>
<div class="section" id="local-gcb-server">
<h2>Local GCB server<a class="headerlink" href="#local-gcb-server" title="Permalink to this headline">¶</a></h2>
<div class="section" id="installation">
<h3>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h3>
<p>First, clone or donload git repository:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">DNKonanov</span><span class="o">/</span><span class="n">GCB</span><span class="o">.</span><span class="n">git</span>
</pre></div>
</div>
<p>Then, install dependencies:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sudo</span> <span class="n">apt</span><span class="o">-</span><span class="n">get</span> <span class="n">install</span> <span class="n">graphviz</span> <span class="n">graphviz</span><span class="o">-</span><span class="n">dev</span> <span class="n">python3</span><span class="o">-</span><span class="n">graphviz</span> <span class="n">python3</span><span class="o">-</span><span class="n">pygraphviz</span>

<span class="n">pip3</span> <span class="n">install</span> <span class="o">-</span><span class="n">r</span> <span class="n">requirements</span><span class="o">.</span><span class="n">txt</span>

<span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">paraslonic</span><span class="o">/</span><span class="n">orthosnake</span>

<span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">DNKonanov</span><span class="o">/</span><span class="n">geneGraph</span>
</pre></div>
</div>
</div>
<div class="section" id="usage">
<h3>Usage<a class="headerlink" href="#usage" title="Permalink to this headline">¶</a></h3>
<p><strong>Add data</strong></p>
<p>To generate a dataset for GCB you need to run <a class="reference external" href="https://github.com/DNKonanov/geneGraph">GeneGraph</a>, a command-line tool to generate graphs and complexity profiles.</p>
<p>Suppose, we have a number of <code class="docutils literal notranslate"><span class="pre">*.fna</span></code> files for 100 different genomes of Mycoplasma.</p>
<p>First, orthogroup file <code class="docutils literal notranslate"><span class="pre">OrthoGroups.txt</span></code> should be generated with <a class="reference external" href="https://github.com/paraslonic/orthosnake">orthosnake</a>. <code class="docutils literal notranslate"><span class="pre">OrthoGroups.txt</span></code> will be in <code class="docutils literal notranslate"><span class="pre">orthosnake/Results</span></code> folder.</p>
<p>Next, use <code class="docutils literal notranslate"><span class="pre">gg.py</span></code> script from geneGraph to generate GCB databases:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python</span> <span class="n">gg</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">i</span> <span class="p">[</span><span class="n">path</span> <span class="n">to</span> <span class="n">OrthoGroups</span><span class="o">.</span><span class="n">txt</span><span class="p">]</span> <span class="o">-</span><span class="n">o</span> <span class="n">Mycoplasma</span>
</pre></div>
</div>
<p>Now move the created <code class="docutils literal notranslate"><span class="pre">Mycoplasma</span></code> folder to <code class="docutils literal notranslate"><span class="pre">GCB/data</span></code>. If there is no <code class="docutils literal notranslate"><span class="pre">data</span></code> folder in <code class="docutils literal notranslate"><span class="pre">GCB</span></code> root folder, just create it with <code class="docutils literal notranslate"><span class="pre">mkdir</span> <span class="pre">data</span></code>.</p>
<p><strong>Start server</strong></p>
<p>To start GCB server on your computer type in GCB_package folder this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python3</span> <span class="n">gcb_server</span><span class="o">.</span><span class="n">py</span>
</pre></div>
</div>
<p>Open <strong>127.0.0.1:8000</strong> or <strong>localhost:8000</strong> in your web-browser and use GCB.</p>
<p>Restart the sever after adding new datasets.</p>
</div>
</div>
<div class="section" id="complete-example">
<h2>Complete example<a class="headerlink" href="#complete-example" title="Permalink to this headline">¶</a></h2>
<p>Comment: we will work in the home directory, consider using other folders according to your preference.</p>
<p><strong>Install conda</strong></p>
<p>If you have not installed conda (miniconda or anaconda) previously, you should first install it. For example, by downloading <a class="reference external" href="https://docs.conda.io/en/latest/miniconda.html">miniconda</a> and running the installation bash script.</p>
<p><strong>Create conda environment for GCB</strong>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">install</span> <span class="o">-</span><span class="n">c</span> <span class="n">conda</span><span class="o">-</span><span class="n">forge</span> <span class="n">mamba</span>
<span class="n">mamba</span> <span class="n">create</span> <span class="o">-</span><span class="n">c</span> <span class="n">conda</span><span class="o">-</span><span class="n">forge</span> <span class="o">-</span><span class="n">c</span> <span class="n">bioconda</span> <span class="o">-</span><span class="n">n</span> <span class="n">gcb</span> <span class="n">snakemake</span>
<span class="n">conda</span> <span class="n">activate</span> <span class="n">gcb</span>
</pre></div>
</div>
<p><strong>Download orthosnake</strong>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~</span>
<span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">paraslonic</span><span class="o">/</span><span class="n">orthosnake</span><span class="o">.</span><span class="n">git</span>
<span class="n">cd</span> <span class="n">orthosnake</span>
</pre></div>
</div>
<p><strong>Download genomes</strong>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">wget</span> <span class="n">ftp</span><span class="p">:</span><span class="o">//</span><span class="n">ftp</span><span class="o">.</span><span class="n">ncbi</span><span class="o">.</span><span class="n">nlm</span><span class="o">.</span><span class="n">nih</span><span class="o">.</span><span class="n">gov</span><span class="o">/</span><span class="n">genomes</span><span class="o">/</span><span class="n">refseq</span><span class="o">/</span><span class="n">assembly_summary_refseq</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>We will download 5 complete genomes:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>grep &quot;Mycoplasma pneumoniae&quot; assembly_summary_refseq.txt | grep &quot;Complete&quot; | awk -F &quot;\t&quot; &#39;{print $20}&#39; | awk &#39;BEGIN{FS=OFS=&quot;/&quot;;filesuffix=&quot;genomic.fna.gz&quot;}{ftpdir=$0;asm=$10;file=asm&quot;_&quot;filesuffix;print ftpdir,file}&#39; &gt; complete_genomes.url
head -5 complete_genomes.url &gt; selected_genomes.url
wget $(cat selected_genomes.url)
gunzip *.gz
mv *.fna fna
</pre></div>
</div>
<p><strong>Run orthofinder</strong></p>
<p>To run orthofinder in 4 threads:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">snakemake</span> <span class="o">-</span><span class="n">j</span> <span class="mi">4</span> <span class="o">--</span><span class="n">use</span><span class="o">-</span><span class="n">conda</span>
</pre></div>
</div>
<p>Installation of prokka and orthofinder (performed during the first run only) and the analysis took 21m on a PC with i7 CPU.</p>
<p>We have inferred orthogroups, they are located in <code class="docutils literal notranslate"><span class="pre">Results/Orthogroups.txt</span></code> file.</p>
<p><strong>Download geneGraph console tools</strong>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~</span>
<span class="n">pip3</span> <span class="n">install</span> <span class="n">gene_graph_lib</span>
<span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">DNKonanov</span><span class="o">/</span><span class="n">geneGraph</span><span class="o">.</span><span class="n">git</span>
</pre></div>
</div>
<p><strong>Build a graph representation of genomes</strong>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mkdir</span> <span class="o">-</span><span class="n">p</span> <span class="n">mpneumoniae</span><span class="o">/</span>
<span class="n">cd</span> <span class="n">mpneumoniae</span><span class="o">/</span>
<span class="n">cp</span> <span class="o">~/</span><span class="n">orthosnake</span><span class="o">/</span><span class="n">Results</span><span class="o">/</span><span class="n">Orthogroups</span><span class="o">.</span><span class="n">txt</span> <span class="o">.</span>
<span class="n">python</span> <span class="o">~/</span><span class="n">geneGraph</span><span class="o">/</span><span class="n">gg</span><span class="o">.</span><span class="n">py</span> <span class="o">-</span><span class="n">i</span> <span class="n">Orthogroups</span><span class="o">.</span><span class="n">txt</span> <span class="o">-</span><span class="n">o</span> <span class="n">mycoplasma_pneumoniae</span>
</pre></div>
</div>
<p>Analysis should take about a minute.</p>
<p><strong>Install GCB local web server</strong>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~</span>
<span class="n">git</span> <span class="n">clone</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">DNKonanov</span><span class="o">/</span><span class="n">GCB</span><span class="o">.</span><span class="n">git</span>
<span class="n">cd</span> <span class="n">GCB</span>
<span class="n">sudo</span> <span class="n">apt</span><span class="o">-</span><span class="n">get</span> <span class="n">install</span> <span class="n">graphviz</span> <span class="n">graphviz</span><span class="o">-</span><span class="n">dev</span> <span class="n">python3</span><span class="o">-</span><span class="n">graphviz</span> <span class="n">python3</span><span class="o">-</span><span class="n">pygraphviz</span>
<span class="n">pip3</span> <span class="n">install</span> <span class="o">-</span><span class="n">r</span> <span class="n">requirements</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<dl class="simple">
<dt><strong>Copy data</strong>::</dt><dd><p>mkdir data
cp -r ~/mpneumoniae/mycoplasma_pneumoniae/ data</p>
</dd>
</dl>
<p><strong>Run local server</strong>:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">python3</span> <span class="n">gcb_server</span><span class="o">.</span><span class="n">py</span>
</pre></div>
</div>
<p>Open <a class="reference external" href="http://127.0.0.1:8000/">http://127.0.0.1:8000/</a> in a web browser and work with GCB.</p>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h3><a href="index.html">Table of Contents</a></h3>
  <ul>
<li><a class="reference internal" href="#">Standalone version</a><ul>
<li><a class="reference internal" href="#command-line-tool">Command-line tool</a><ul>
<li><a class="reference internal" href="#roadmap">Roadmap</a></li>
<li><a class="reference internal" href="#prerequisites">Prerequisites</a></li>
<li><a class="reference internal" href="#orthogroup-inference">Orthogroup inference</a></li>
<li><a class="reference internal" href="#building-a-graph-and-complexity-estimation-with-a-single-command">Building a graph and complexity estimation with a single command</a></li>
</ul>
</li>
<li><a class="reference internal" href="#local-gcb-server">Local GCB server</a><ul>
<li><a class="reference internal" href="#installation">Installation</a></li>
<li><a class="reference internal" href="#usage">Usage</a></li>
</ul>
</li>
<li><a class="reference internal" href="#complete-example">Complete example</a></li>
</ul>
</li>
</ul>

  <h4>Previous topic</h4>
  <p class="topless"><a href="interface.html"
                        title="previous chapter">GUI description</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="FAQ.html"
                        title="next chapter">Frequently Asked Questions</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/standalone.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="FAQ.html" title="Frequently Asked Questions"
             >next</a> |</li>
        <li class="right" >
          <a href="interface.html" title="GUI description"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">Genome Complexity Browser  documentation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2020, A. Manolov, D. Konanov, D. Fedorov, R. Vereshchagin, E. Ilina.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 2.2.0.
    </div>
  </body>
</html>